### PR TITLE
[#9075] - Remove any unnecessary dependency from distribution

### LINKED ISSUE #9075: Remove any unnecessary dependency from distribution
### Description

Not only to reduce the distribution size but also to reduce the vulnerability surface, we should review the dependencies included in the distribution and keep only those strictly necessary to runtime and re-augmentation.

As the server is a mutable jar, these additional dependencies are a trade-off as part of continuous testing support from Quarkus. However, in theory, these dependencies won't actually be loaded in the runtime application.

### Discussion

_No response_

### Motivation

Not only to reduce the distribution size but also to reduce the vulnerability surface, we should review the dependencies included in the distribution and keep only those strictly necessary to runtime and re-augmentation.

### Details

Looks like the best approach should be to exclude these dependencies through Maven (instead of using Quarkus properties such as `quarkus.class-loading.removed-artifacts`).

The proposal is to have a specific profile in the root pom that explicitly excludes the unwanted dependencies. The trade-off here is that we need to make sure the distribution is built using this profile.

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: quarkus/pom.xml [hunk 1]
```diff
@@ -169,5 +169,105 @@
...
+        <profile>
+            <!--
+                This profile is intended for building the distribution.
+                It should be enabled whenever building the distribution so that only the expected
+                artifacts are included within the distribution.
+                By
+            -->
+            <id>distribution</id>
+            <dependencyManagement>
+                <dependencies>
```
Comment (stianst) [inline quarkus/pom.xml:183]: I'm not really following this - why would there be a "distribution" profile, and not just remove the deps always?

Comment (pedroigor) [reply quarkus/pom.xml:183]: When running tests we don't want to completely disable the test capabilities from quarkus such as dev services / testcontainers.

Comment (stianst) [reply quarkus/pom.xml]: In that case wouldn't you just use the test scope for the dependency instead of a profile?

**Comment by pedroigor** [comment]
Closing for now as we have some work required from Quarkus to properly remove dependencies for any extension we use.

**Pull request closed without merging**